proposals/0052-svg-components.md

```diff
@@ -0,0 +1,136 @@
+
+Previously, `astro-icon` was introduced to experiment with `.svg` support but, with the relase of `astro:assets`, there's now an opportunity to provide optimized `.svg` imports that simplify the developer experience and reduce client-side bloat.
+
+The core motivation is to automate best practices around `.svg` usage, enabled developers to effortlessly use and optimize their `.svg` assets in Astro projects.
+
+# Goals
+
+- Enable importing and rendering `.svg` files as Astro components.
+- Enable importing and rendering `.svg` files in MDX files
+- Allow users to pass props to the root `svg` element while allowing for prop overrides.
+- Maintain backwards compatibility with existing `.svg` import behavior.
+- Strip unnecessary attributes like `xmlns` and `version` from `.svg` files to reduce size.
+- Allow third-party plugins to hook into the `.svg` import behavior to generate their own component libraries.
+
+# Non-Goals
+
+- Future: Allow users to opt-in to optimized `.svg` rendering by utilizing `symbol` and `use` for repeated instances.
+- Future: Adding support for `.svg` files as framework components.
+- Out-of-scope: Advanced tree-shaking or SSR optimizations for large icon sets.
+- Out-of-scope: Advanced `.svg` file optimization such as `svgo`.
+- Non-goal: Adding support for `.json` icon sets such as `@iconify/json`.
+- Non-goal: SVG content analysis for accessibility.
+
+# Detailed Design
+
+## Importing SVG Files as Components
+
+The core design of this proposal allows `.svg` files to be imported and used as components in Astro. This means developers can import `.svg` files into their Astro components and render them directly as if they were Astro components. The goal is to maintain ease of use while automating performance optimizations under the hood.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<Logo />
+```
+
+This would generate the following output:
+
+```astro
+<svg width="24" height="24">
+    <!-- SVG Content -->
+</svg>
+```
+
+## Prop Inheritance and Overriding
+
+When a `.svg` is treated as a component, users can pass attributes such as `width`, `height`, `fill`, `stroke`, and other common SVG attributes. These attributes will automatically be applied to the root `<svg>` element unless they conflict with existing properties in the original `.svg` file, in which case the passed props will override the existing ones.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg'
+---
+
+<Logo />
+
+<!-- Pass width and height to override default size -->
+<Logo width={100} height={100} fill="blue" />
+```
+
+This would generate the following output:
+
+```astro
+<svg width="24" height="24">
+    <!-- SVG Content -->
+</svg>
+<svg width="100" height="100" fill="blue">
+    <!-- SVG Content -->
+</svg>
+```
+
+## Accessibility Considerations
+
+By default Astro allows you to configure the SVG to meet your accessibility needs.
```

Comment:
Does that mean Astro does nothing special? Maybe it would be clearer to say it's on the user to make it accessible